staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields.go

```diff
@@ -65,12 +65,13 @@ func NewManaged(f fieldpath.ManagedFields, t map[string]*metav1.Time) ManagedInt
 // RemoveObjectManagedFields removes the ManagedFields from the object
 // before we merge so that it doesn't appear in the ManagedFields
 // recursively.
-func RemoveObjectManagedFields(obj runtime.Object) {
+func RemoveObjectManagedFields(obj runtime.Object) error {
 	accessor, err := meta.Accessor(obj)
 	if err != nil {
-		panic(fmt.Sprintf("couldn't get accessor: %v", err))
+		return fmt.Errorf("couldn't get accessor: %v", err)
```

Comment:
yeah, this can't and shouldn't happen. I'd be happy with a more explicit comment along with the panic.

Alternately, we could just ignore it (if we can't get an accessor, then we probably don't have any managed fields to set to nil), but I really don't think it'd be any better than what we have today.